### PR TITLE
[backward incompatible] support arbitrary settings in unbound.conf(5)

### LINKED ISSUE #19: Empty chroot configuration in unbound.conf
Even if `unbound_config_chroot` is empty it generates `chroot: ""` line in unbound.conf

### LINKED ISSUE #21: "Register unbound version" task is skipped in check-mode
as a result `ansible-playbook -C` fails

### PR DESCRIPTION
**Pull request by trombik**
this PR overhauls logic to create `unbound.conf(5)`. the previous implementation hard-coded settings in the template. as a result, whenever users needed new configurations, the template must have been modified. an example is #23. this PR enables users to set arbitrary settings (but see #28 ). some settings remain hard-coded because the role has to do something depending on their values. unlike other roles, no default is provided mostly because `unbound.conf(5)` is not valid YAML, which makes it impossible to merge user-defined dict into default dict. also, closes some issues.

- closes #22 
- closes #23 
- closes #20 
- fixes #19 
- fixes #21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,78 +18,164 @@ None
...
-| unbound\_config\_remote\_control\_control\_enable | `control-enable` | yes |
-| unbound\_config\_remote\_control\_control\_use\_cert | `control-use-cert` | no |
-| unbound\_config\_remote\_control\_control\_interface | `control-interface` | "" |
-| unbound\_config\_server\_key\_file | `server-key-file` | {{ unbound\_config\_directory }}/unbound\_server.key |
-| unbound\_config\_server\_cert\_file | `server-cert-file` | {{ unbound\_config\_directory }}/unbound\_server.pem |
-| unbound\_config\_control\_key\_file | `control-key-file` | {{ unbound\_config\_directory }}/unbound\_control.key |
-| unbound\_config\_control\_cert\_file | `control-cert-file` | {{ unbound\_config\_directory }}/unbound\_control.pem |
-| unbound\_forward\_zone | `forward-zone` | [] |
+| `unbound_user` | user of `unbound` | `{{ __unbound_user }}` |
+| `unbound_group` | group of `unboun` | `{{ __unbound_group }}` |
```
Comment (mitsururike) [inline README.md]: typo: `unboun` -> `unbound`

File: README.md [hunk 2]
```diff
@@ -25,78 +18,164 @@ None
...
+| `unbound_service` | service name of `unbound` | `unbound` |
+| `unbound_conf_dir` | path to config directory | `{{ __unbound_conf_dir }}` |
+| `unbound_conf_file` | path to `unbound.conf(5)` | `{{ __unbound_conf_dir }}/unbound.conf` |
```
Comment (mitsururike) [inline README.md]: s/__unbound_conf_dir/unbound_conf_dir/ or users must modify unbound_conf_file when unbound_conf_dir is set.

File: README.md [hunk 3]
```diff
@@ -25,78 +18,164 @@ None
...
+  access-control: 127.0.0.0/8 allow
+```
+
+## `unbound_forward_zone`
+
+`unbound_forward_zone` is a list of zones described in "Forward Zone Options"
+in `unbound.conf(5)`.
+
+An element is a dict of zone. The dict must have a mandatory key, `name`, whose
+value is the name of the zone . Other keys in the dict is a setting for the
```
Comment (mitsururike) [inline README.md]: grammar: `keys` -> `key` or `is a` -> `are`

File: README.md [hunk 4]
```diff
@@ -25,78 +18,164 @@ None
...
+zone, such as `forward-addr`, whose value is the value of the setting. See
+"Forward Zone Options" in `unbound.conf(5)`. Values of optional settings can be
```
Comment (mitsururike) [inline README.md]: grammar: `Values` -> `Value`

File: tasks/deprecate.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+- assert:
+    that:
+      - not unbound_config_private_domain is defined
+    msg: unbound_config_private_domain is deprecated, use unbound_config_server instead.
+
+- assert:
+    that:
+      - not unbound_config_private_address is defined
+    msg: unbound_config_private_address is deprecated, use unbound_config_server instead.
```
Comment (mitsururike) [inline tasks/deprecate.yml]: with_items:
  - unbound_config_outgoing_interface 
  - unbound_config_...
  - ...
does not work?

Comment (trombik) [inline tasks/deprecate.yml]: the variable name will be expanded to its value, and users will not see which variable should be removed.

Comment (mitsururike) [inline tasks/deprecate.yml]: it is not expanded to its value.
```
       failed: [localhost] (item=unbound_config_private_domain) => {
           "assertion": "not unbound_config_private_domain is defined", 
           "evaluated_to": false, 
           "failed": true, 
           "item": "unbound_config_private_domain", 
           "msg": "unbound_config_private_domain is deprecated, use unbound_config_server instead."
       }
```

Comment (trombik) [inline tasks/deprecate.yml]: correct. the trick was not to use array in `that`, which is something i got used to when using `assert`.

**Pull request merged**